src/document/node.js

```diff
@@ -232,19 +232,14 @@ CKEDITOR.define( [ 'document/attribute', 'utils', 'ckeditorerror' ], ( Attribute
 		}
 
 		/**
-		 * Gets the number of attributes.
+		 * Returns attribute iterator. It can be use to create a new element with the same attributes:
 		 *
-		 * @protected
-		 * @returns {Number} Number of attributes.
+		 *	  const copy = new Element( element.name, element.getAttrIterator() );
+		 *
+		 * @returns {Iterable.<document.Attribute>} Attribute iterator.
 		 */
-		_getAttrCount() {
-			let count = 0;
-
-			for ( let attr of this._attrs ) { // jshint ignore:line
-				count++;
-			}
-
-			return count;
+		getAttrIterator() {
```

Comment:
Note for me in the future: it's better to have an iterator if attrs are private because we can change them to map or set or whatever and all these types have the same common API – they are iterable. We could consider always returning an array, because, even as the API docs state, you will normally access the entire set. However, after starting keeping attr in map or set we would be creating an array everytime, even though you don't need it. On the other hand, `Array.from( attrsMap )` may be faster than `Array.from( node.getAttrs() )` when `getAttrs()` returns iterable, but it's too early to think about it.